specs/execution-summary.md

```diff
@@ -0,0 +1,178 @@
+      - Request URL
+      - List of unique request log messages for this request issued by the proxy, grouped and sorted by severity. Excluding the `InterceptedRequest` and `PassedThrough` severity.
+  - if grouped by message type:
+    - List of severities sorted by severity level. For each severity:
+      - Severity name
+      - List of unique URLs logged for this severity sorted alphabetically
+- Summary
+  - Total number of requests captured
+  - For each type of request log message:
+    - Number of requests with this type of request log message
+
+### Sample summary
+
+#### Grouped by URL
+
+```md
+# Microsoft Graph Developer Proxy execution summary
+
+Date: 2023-02-06 12:00:00
+
+## Requests
+
+### GET https://graph.microsoft.com/v1.0/me
+
+#### Warning
+
+- To improve performance of your application, use the $select parameter. ore info at https://learn.microsoft.com/graph/query-parameters#select-parameter
+
+#### Tip
+
+- To handle API errors more easily, use the Graph SDK. More info at https://aka.ms/move-to-graph-js-sdk
+
+#### Failed
+
+- Calling https://graph.microsoft.com/v1.0/me again before waiting for the Retry-After period. Request will be throttled
+
+#### Chaos
+
+- 503 ServiceUnavailable
+- 429 TooManyRequests
+- 500 InternalServerError
+
+#### Mocked
+
+### GET https://graph.microsoft.com/v1.0/me/users
+
+#### Warning
+
+- To improve performance of your application, use the $select parameter. ore info at https://learn.microsoft.com/graph/query-parameters#select-parameter
+
+#### Tip
+
+- To handle API errors more easily, use the Graph SDK. More info at https://aka.ms/move-to-graph-js-sdk
+
+#### Failed
+
+- Calling https://graph.microsoft.com/v1.0/me/users again before waiting for the Retry-After period. Request will be throttled
+
+#### Chaos
+
+- 503 ServiceUnavailable
+- 429 TooManyRequests
+- 500 InternalServerError
+
+#### Mocked
+
+## Summary
+
+Category|Count
+--------|----:
+Requests intercepted|100
+Requests passed through|10
+Requests with chaos|80
+Requests mocked|10
+Tips|120
+Warnings|30
+Failures|5
+```
+
+#### Grouped by message type
+
+```md
+# Microsoft Graph Developer Proxy execution summary
+
+Date: 2023-02-06 12:00:00
+
+## Message types
+
+### Requests intercepted
+
+- GET https://graph.microsoft.com/v1.0/me
+- GET https://graph.microsoft.com/v1.0/me/messages
+- GET https://graph.microsoft.com/v1.0/me/users
+
+### Requests passed through
+
+- GET https://graph.microsoft.com/v1.0/me
+
+### Requests with chaos
+
+#### 429 TooManyRequests
+
+- GET https://graph.microsoft.com/v1.0/me
+- GET https://graph.microsoft.com/v1.0/me/users
+  
+#### 500 InternalServerError
+
+- GET https://graph.microsoft.com/v1.0/me
+
+### Requests mocked
+
+#### https://graph.microsoft.com/v1.0/me/*
+
+- GET https://graph.microsoft.com/v1.0/me/messages
+
+### Tips
+
+#### To handle API errors more easily, use the Graph SDK. More info at https://aka.ms/move-to-graph-js-sdk
+
+- GET https://graph.microsoft.com/v1.0/me
```

Comment:
Same here, per method / URL we should know the # of unique instances of this TIP (or warning, failure, etc.)

Reply:
Would the same approach, with the count of instances displayed between `()`, work here as well?

```md
## Message types

### Requests intercepted

- GET https://graph.microsoft.com/v1.0/me (10)
- GET https://graph.microsoft.com/v1.0/me/messages (15)
- GET https://graph.microsoft.com/v1.0/me/users (12)
```

Reply:
I'd prepend it for context.